packages/react-query-devtools/src/devtools.tsx

```diff
@@ -40,11 +40,12 @@ export function ReactQueryDevtools(
   props: DevtoolsOptions,
 ): React.ReactElement | null {
   const queryClient = useQueryClient()
+  const client = props.client || queryClient
```

Comment:
`useQueryClient` can handle this internally, so we can do:

```
const client = useQueryClient(props.client)
```

`useQueryClient` will read from context or return the passed prop if it isn't undefined:

https://github.com/TanStack/query/blob/e61c7849b1f4dd5bd7909dc2204e7c152c1b9479/packages/react-query/src/QueryClientProvider.tsx#L10-L22